src/Microsoft.Data.Analysis/DataFrame.cs

```diff
@@ -485,7 +487,30 @@ public DataFrame Append(IEnumerable<object> row = null, bool inPlace = false)
                     }
                     if (value != null)
                     {
-                        value = Convert.ChangeType(value, column.DataType);
+                        try
+                        {
+                            value = Convert.ChangeType(value, column.DataType, cultureInfo);
+                        }
+                        catch(FormatException)
+                        {
+                            Console.Write($"Value \"{value}\" cannot be converted to type {column.DataType} (Column name: {column.Name}). ");
+
+                            if (column.DataType == typeof(double))
+                            {
+                                Console.WriteLine("Converting to Double.NaN instead.");
+                                value = Double.NaN;
```

Comment:
I like this change, but do you think you could separate it out into it's own PR? Reason is that I don't think we've tested any of our existing APIs with `NaN` values. So I'd like to get coverage by modifying `DataFrameTests.cs:MakeDataFrameWithNumericColumns` generate a row with `Double.NaN` and `Single.NaN`. That should automatically give us coverage over all the APIs